### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by dbarbier**

### CODE REVIEW

File: tsfresh/utilities/distribution.py [hunk 1]
```diff
@@ -319,7 +319,9 @@ def distribute(self, func, partitioned_chunks, kwargs):
         :return: The result of the calculation as a list - each item should be the result of the application of func
             to a single element.
         """
-
+        if isinstance(partitioned_chunks, Iterable):
+            # since dask 2.0.0 client map no longer accepts iterables
+            partitioned_chunks = list(partitioned_chunks)
```
Comment (MaxBenChrist) [inline tsfresh/utilities/distribution.py]: Doesn't this casting (`partitioned_chunks = list(partitioned_chunks)`) creates a risk to blow up the memory?

Let us say we have data that is 6 GB big. Now, partitioned_chunks gets casted as list, so we have another 6 GB, 12 used in total. Now, we push that data into the workers and suddenly we use 18 GB of memory in total.

On the other hand, the `client.map` will break on the iterable, right?

Comment (dbarbier) [reply tsfresh/utilities/distribution.py]: True, but this is the exact same fix you applied at #550, and I like consistency ;)

Comment (dbarbier) [reply tsfresh/utilities/distribution.py]: In fact I was wrong, client.map internally converts iterables into a list, so there is no change.
What you want is a queue of fixed size, but it is harder to implement.

**Comment by nils-braun** [comment]
@MaxBenChrist lgtm

**Pull request merged**